TeamSnapSDK/SDK/CollectionJSON/TSDKCollectionObject.m

```diff
@@ -754,6 +762,7 @@ - (void)saveWithURL:(NSURL *)url completion:(TSDKSaveCompletionBlock)completion
                 if ([objects.collection isKindOfClass:[NSArray class]]) {
                     NSArray *returnedCollections = (NSArray *)objects.collection;
                     if ([returnedCollections count]==1) {
+                        weakSelf._objectIdentifier = nil;
                         [weakSelf setCollection:[returnedCollections objectAtIndex:0]];
```

Comment:
Could we have `setCollection:` implicitly nullify `_objectIdentifier`?

Reply:
Yes, although I don't think that would have much larger reach than this would. the tricky part that I haven't set up yet is when collection.data["id'] is set (I only found one instance of it in the app so far, and it was safe.. but I worry about the future)

Reply:
I'm thinking more of adding new methods where someone forgets to manually nullify this property.